docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+    // Calculate the year
+    uint32_t num_years = 0;
+    while (true) {
+        uint32_t seconds_in_year = (num_years % 4 == 0 && (num_years % 100 != 0 || num_years % 400 == 0)) ? seconds_per_leap_year : seconds_per_year;
+        if (remaining_seconds < seconds_in_year) {
+            year = epoch_years + num_years;
+            break;
+        }
+        remaining_seconds -= seconds_in_year;
+        num_years++;
+    }
+
+    // Calculate the month
+    uint32_t num_days = 0;
+    for (uint8_t i = 1; i <= 13; i++) {
+        uint32_t days_in_month = 0;
+        if (i == 2) {
+            if (year % 4 == 0 && (year % 100 != 0 || year % 400 == 0)) {
+                days_in_month = 29;
+            } else {
+                days_in_month = 28;
+            }
+        } else if (i == 4 || i == 6 || i == 9 || i == 11) {
+            days_in_month = 30;
+        } else {
+            days_in_month = 31;
+        }
+        uint32_t seconds_in_month = days_in_month * seconds_per_day;
+        if (remaining_seconds < seconds_in_month) {
+            month = i;
+            break;
+        }
+        remaining_seconds -= seconds_in_month;
+        num_days += days_in_month;
+    }
+
+    // Calculate the day
+    day = remaining_seconds / seconds_per_day + 1;
+    remaining_seconds = remaining_seconds % seconds_per_day;
+
+    // Calculate the hour
+    hour = remaining_seconds / seconds_per_hour;
+    remaining_seconds = remaining_seconds % seconds_per_hour;
+
+    // Calculate the minute
+    minute = remaining_seconds / seconds_per_minute;
+
+    // Calculate the second
+    second = remaining_seconds % seconds_per_minute;
+}
+```
+
+Build the smart contract again and deploy it to the local node as you did previously.
+
+Now you can send an `hi` action to the local node and observe that both `hi` and `gettime` actions are executed.
+
+```shell
+dune --send-action hello hi '[ama]' hello@active
+```
+
+```txt
+executed transaction: 78ece3e63c3c6f11a210a9229f32b822c825fc72b268b1325b7dc1798797f460  104 bytes  431 us
+warning: transaction executed locally, but may not be confirmed by the network yet         ] 
+#         hello <= hello::hi                    {"nm":"ama"}
+>> Name : ama
+#         hello <= hello::gettime               ""
+>> Year: 2023, Month: 2, Day: 24, Hour: 0, Minute: 42, Second: 5
+```
+
+## Built-in Types
+
+Antelope supports several C++ data types for developing smart contracts. Developers can use these types to define data structures and write functions that interact with the EOS blockchain and smart contract system. Here are some of the most commonly used types:
+
+- `uint64_t`: This is an unsigned 64-bit integer type used for storing numeric values. It's commonly used for representing amounts of currency or other assets.
+- `name`: This is a type that represents an account name on the EOSIO network. Account names are 12-character strings that uniquely identify user accounts.
```

Comment:
Should state that `name`s are numbers underneath so that `.value()` later makes sense.

Reply:
fixed